Lib/enum.py

```diff
@@ -965,38 +1223,39 @@ def __contains__(self, other):
             raise TypeError(
                 "unsupported operand type(s) for 'in': '%s' and '%s'" % (
                     type(other).__qualname__, self.__class__.__qualname__))
+        if other._value_ == 0 or self._value_ == 0:
+            return False
         return other._value_ & self._value_ == other._value_
 
     def __iter__(self):
         """
         Returns flags in decreasing value order.
         """
-        members, extra_flags = _decompose(self.__class__, self.value)
-        return (m for m in members if m._value_ != 0)
+        val = self._value_
+        while val:
+            i = 2 ** (val.bit_length() - 1)
+            member = self._value2member_map_.get(i)
+            if member is not None:
+                yield member
```

Comment:
Changed.

Reply:
the code looks the same.  I was wondering if it could be:

```python3
val = self._value_ & self._flag_mask_
while val:
    i = 2 ** (val.bit_length() - 1)
    yield self._value2member_map_.get(i)
    val ^= i
```